docs/4-cloud-computing/4.3.7-app-service.md

```diff
@@ -0,0 +1,92 @@
+      description: Create A Web App and Scale and Monitor
+      estMinutes: 240
+      technologies:
+      - Azure
+      - Azure App Service
+      - Azure CLI
+---
+# Azure App Service
+
+Azure App Service is a fully managed platform for building, deploying, and scaling web apps. It supports various programming languages and frameworks like .NET, Java, Node.js, Python, and more, allowing developers to build web and mobile back-ends with ease. Through Azure App Service, you can integrate with various Azure tools like Azure DevOps for CI/CD and take advantage of built-in features such as custom domains and SSL certificates. It's designed with best practices in mind, providing scalability, robust monitoring, and security, thereby enabling a seamless and efficient app hosting solution.
+
+> [Azure App Service](https://learn.microsoft.com/en-us/azure/app-service/) Documentation
+
+### Exercise 1: Deploy a Java Application
+
+We will be using Liatrio's version of the spring-petclinic for this exercise, it's a simple java application that deploys a web app for a virtual pet clinic.
+
+1. Fork [Spring-Petclinic](https://github.com/liatrio/spring-petclinic.git). (We will be using this later for Continuous Deployment)
+
+2. Clone your new fork.
+
+```bash
+git clone https://github.com/YOUR_NAME/spring-petclinic.git
+```
+
+!> There is an official repo for the spring-petclinic, we will not be using that since it contains different build tests that may not be useful for our purposes and might break the application on build.
+
+3. Install the maven project and ensure that you can reach the application locally to move onto the next step.
+
+![Spring Petclinic Landing Page](img4/spring-petclinic-1.png ':size=912x400')
+
+#### Deploy the Java App to App Service
+
+1. Read the above documentation about Azure App Service and use Microsofts official [Maven Plugin](https://learn.microsoft.com/en-us/azure/app-service/quickstart-java?tabs=javase&pivots=platform-linux-development-environment-maven) to deploy a java application.
+2. Make sure the application you deploy can be reached by copying the domain it gives you at the end and pasting that into your web browser.
+
+?> If you wish, you may change the name of your application in the pom file before you deploy.
+
+### Exercise 2: Deploy a Docker Application
+
+We have seen that the Azure App Service can be useful for deploying a Java application using Azures built in maven plugin, but what if our application doesn't use java? What if it uses something that isn't natively supported by Azure? That's where docker containers come in, they are super versatile and can run a wide range of applications that are otherwise not supported by the Azure App Service natively. In this exercise, we will be taking the spring-petclinic and publish a docker image to be used with the Azure App Service. This will demonstrate a more practical use for App Service, as a majority of Engineers would be using Docker Containers.
```

Comment:
```suggestion
We have seen that the Azure App Service can be useful for deploying a Java application using Azures built in maven plugin, but what if our application doesn't use Java? What if it uses something that isn't natively supported by Azure? That's where docker containers come in, they are super versatile and can run a wide range of applications that are otherwise not supported by the Azure App Service natively. In this exercise, we will be taking the spring-petclinic and publish a Docker image to be used with the Azure App Service. This demonstrates a more practical use for App Service with the rise of containerization.
```